testsuite/tests/effects/test_lazy.ml

```diff
@@ -0,0 +1,34 @@
+(* TEST *)
+
+effect Stop : unit
+
+let f count =
+  let _n = (Domain.self ():> int) in
+  let r = ref 0 in
+  for i = 1 to count do
+    incr r;
+    if i = count / 2 then perform Stop
+  done;
+  !r
+
+let _ =
+  let l = lazy (f 1_000) in
+  let v1 =
+    try Lazy.force l with
+    effect Stop k -> continue k () in
+    Printf.printf "%d\n" v1;
+  let l2 = lazy (f 1_000) in
```

Comment:
Might be nice to make this a different number, e.g. `2_000`, so that things in the output tie unambiguously back to the test code.